pkg/apis/pipeline/v1beta1/pipelinerun_validation_test.go

```diff
@@ -360,6 +544,7 @@ func TestPipelineRun_Validate(t *testing.T) {
 						Params: []v1beta1.Param{{
 							Name: "task-words",
 							Value: v1beta1.ParamValue{
+								Type:     v1beta1.ParamTypeArray,
```

Comment:
why is type array added in some of these test cases? it doesn't seem related to the changes in this commit -- would rather not add them in this commit if not necessary

Reply:
That's because Chuang pointed out that we should be ParamType checking as well when we are propagating parameters. I had overlooked it and so my original test cases didn't have that. When I added the Type checking in line 189 of `pipelinerun_validation.go` to do it properly for object params, the arrays tests failed as well. Hence this change.

Reply:
Let me know.